src/utils/scorers.ts

```diff
@@ -176,10 +178,52 @@ export const deleteScorer = async (scorerId: string): Promise<void> => {
  */
 export const createCodeScorerVersion = async (
   scorerId: string,
-  codeContent: string
+  codeContent: string,
+  validationResult?: string
 ): Promise<ScorerVersion> => {
   const client = new GalileoApiClient();
   await client.init();
 
-  return await client.createCodeScorerVersion(scorerId, codeContent);
+  return await client.createCodeScorerVersion(
+    scorerId,
+    codeContent,
+    validationResult
+  );
+};
+
+/**
+ * Validates code scorer content and waits for the result.
+ * This function submits the code for validation and polls until complete or timeout.
+ *
+ * @param codeContent - The Python code content to validate.
+ * @param scoreableNodeTypes - The node types that this scorer can score.
+ * @param timeoutMs - Maximum time to wait for validation (default: 60000ms).
+ * @param pollIntervalMs - Interval between polling attempts (default: 1000ms).
+ * @returns A promise that resolves to the validation result.
+ * @throws Error if validation fails or times out.
+ *
+ * @example
+ * ```typescript
+ * const result = await validateCodeScorer(
+ *   'def score(input, output): return 1.0',
+ *   [StepType.llm]
+ * );
+ * console.log(result.result.score_type);
+ * ```
+ */
+export const validateCodeScorer = async (
+  codeContent: string,
+  scoreableNodeTypes: StepType[],
+  timeoutMs?: number,
+  pollIntervalMs?: number
+): Promise<ValidateRegisteredScorerResult> => {
+  const client = new GalileoApiClient();
+  await client.init();
+
+  return await client.validateCodeScorerAndWait(
```

Comment:
does this happen before the scorer is created?  if not does the scorer get cleaned up if the validation fails?